be/src/storage/compaction_manager.cpp

```diff
@@ -561,8 +560,32 @@ std::unordered_set<CompactionTask*> CompactionManager::get_running_task(const Ta
     return res;
 }
 
+int32_t CompactionManager::compute_max_compaction_concurrency() const {
+    int32_t max_task_num = 0;
+    // new compaction framework
+    if (config::base_compaction_num_threads_per_disk >= 0 && config::cumulative_compaction_num_threads_per_disk >= 0) {
```

Comment:
In original olap_server code, it is initialized in main thread, there is no concurrency issue. However, now moved to CompactionManager to support dynamic configvar change.

So direct reference to `config::max_compaction_concurrency` and `config::cumulative_compaction_num_threads_per_disk` and `config::base_compaction_num_threads_per_disk` is in a risk, if they are mutable.

A better approach is to define a local var to capture its value and use the local var to do the subsequent calculation.

Reply:
tks review,I've optimized it here.
But config::cumulative_compaction_num_threads_per_disk and config::base_compaction_num_threads_per_disk 
don't support dynamic update.